common/meta/types-v1/src/example.rs

```diff
@@ -0,0 +1,122 @@
+//  Copyright 2022 Datafuse Labs.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+use common_exception::Result;
+use serde::Deserialize;
+use serde::Serialize;
+
+// This is an example for versioned struct and backward compatible with the old versions:
+// 1. Write ser<V1> to metasrv: <key, ser<V1>>
+// 2. Upgrade V1 to a new struct V2 with a new filed of the V2
+// 3. Write ser<V2> to metasrv: <key, ser<V2>>
+// 4. Read ser<V1> data from metasrv, cast it to V2
+// 5. Read ser<V2> data from metasrv, return V2
+
+// Rules:
+// 1. All the type struct is immutable
+// 2. If you want add/remove/modify the type struct must create a new struct(it's immutable too)
+// 3. Impl From<V1> for V2
+// 4. Impl TryFrom<String> for V2, get the V2 serialize string
+// 5. Impl TryFrom<V2> for String, get the deserialize V2 from string
+
+// Version wrapper for Serialize/Deserialize.
+// The serialized bytes will put into meta through KV API:
+// <key, ser<VersionExample>>
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "version")]
+enum VersionedExample {
+    #[serde(rename = "v1")]
+    V1(ExampleV1),
+    #[serde(rename = "v2")]
+    V2(ExampleV2),
+}
+
+// V1(please don't modify this struct).
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct ExampleV1 {
+    pub tenant: String,
+}
+
+// From V1 to string.
```

Comment:
This converter is for demonstration only, right?
Seems that, in practice the conversions should  be 
- `ExampleVn ~> Result<Vec<u8>>` 
-  `Vec<u8> ~> Result<ExampleVn>`